### DISCUSSION TITLE
Constrain grammar-constrained speech-recognition to whole words?

### DISCUSSION THREAD

**Discussion by ulatekh** [ggml-org/whisper.cpp#2047] [General]
After using whisper.cpp with a grammar for some time, I've finally realized a pattern to one of its erroneous behaviors...it's matching partial words at the end of valid phrases. For instance, it may return "ei" when a valid word at that point was "eight", or "hi" when a valid word at that point would be "high". Is there any way to constrain it to full words specified in the grammar? If not, can anyone tell me where to start digging in the source code?

**Comment by philipag** [comment]
same as #2496
